ui/src/app/dashboard/dashboard.component.ts

```diff
@@ -11,17 +14,47 @@ import {JobStatus} from "../shared/model/JobStatus";
 export class DashboardComponent implements OnInit {
   aggregationResponse: AggregationResponse;
   statusArray: Array<JobStatus> = [JobStatus.Succeeded, JobStatus.Aborted, JobStatus.Running, JobStatus.Failed];
+  timeFrameToDescriptionMap = timeFrameToDescriptionMap;
 
-  constructor(private readonly activatedRoute: ActivatedRoute) {}
+  selectedTimeFrame = defaultTimeFrame;
+  timeFrames: Array<TimeFrame> = [TimeFrame.HOURS1, TimeFrame.HOURS8, TimeFrame.HOURS24,
+    TimeFrame.DAYS7, TimeFrame.DAYS30, TimeFrame.ALLTIME];
+
+  constructor(private readonly router: Router,
+              private readonly activatedRoute: ActivatedRoute) {}
 
   ngOnInit() {
-    this.aggregationResponse = this.activatedRoute.snapshot.data['aggregations'];
+    this.activatedRoute.queryParams.subscribe(() => this.reloadAggregations());
+
+    // make the default toggle option match to url param
+    if (this.activatedRoute.snapshot.queryParams['timeFrame'] != null) {
```

Comment:
Shorter and handles the empty string case `if (this.activatedRoute.snapshot.queryParams['timeFrame']) { ...`

Reply:
Nice!